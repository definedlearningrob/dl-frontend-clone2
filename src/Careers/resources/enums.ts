export enum CourseType {
  HIGH_SCHOOL = 'HIGH_SCHOOL',
  MIDDLE_SCHOOL = 'MIDDLE_SCHOOL',
}
export enum TagTypes {
  SYSTEM = 'SYSTEM',
  ENTITY = 'ENTITY',
}

export enum ResourceClass {
  ASSIGNMENT_SUBMISSION_FILE = 'ASSIGNMENT_SUBMISSION_FILE',
  ATTACHMENT = 'ATTACHMENT',
  CATALOG = 'CATALOG',
  COURSE = 'COURSE',
  EXTENSION_FIELD = 'EXTENSION_FIELD',
  EXTENSION_FIELD_FILE = 'EXTENSION_FIELD_FILE',
  LESSON = 'LESSON',
  PORTFOLIO_PROJECT = 'PORTFOLIO_PROJECT',
  PORTFOLIO_PROJECT_FILE = 'PORTFOLIO_PROJECT_FILE',
  PUBLIC_RESOURCE = 'PUBLIC_RESOURCE',
  RESUME = 'RESUME',
  SLIDE_BACKGROUND_IMAGE = 'SLIDE_BACKGROUND_IMAGE',
  SLIDE_IMAGE = 'SLIDE_IMAGE',
  SLIDE_VIDEO = 'SLIDE_VIDEO',
  TASK = 'TASK',
  TASK_FILE = 'TASK_FILE',
  TRACK = 'TRACK',
  UNIT = 'UNIT',
  VIDEO = 'VIDEO',
  BADGE = 'BADGE',
  ENTITY = 'ENTITY',
}

export enum AssetType {
  FILE = 'FILE',
  IMAGE = 'IMAGE',
  VIDEO = 'VIDEO',
  DC_LOGO = 'DC_LOGO',
  DC_ICON = 'DC_ICON',
  DL_LOGO = 'DL_LOGO',
  DL_ICON = 'DL_ICON',
}

export enum AssessmentType {
  HIGH_SCHOOL = 'HIGH_SCHOOL',
  MIDDLE_SCHOOL = 'MIDDLE_SCHOOL',
}

export enum AssessmentStatuses {
  COMPLETED = 'COMPLETED',
  FAILED = 'FAILED',
  FINISHED = 'FINISHED',
  IN_PROGRESS = 'IN_PROGRESS',
  NOT_STARTED = 'NOT_STARTED',
  PROCESSING_RESULTS = 'PROCESSING_RESULTS',
}

export enum ConversationContextTypes {
  ASSIGNMENT = 'ASSIGNMENT',
  LESSON = 'LESSON',
  COURSE = 'COURSE',
  CHECK_IN_ANSWER = 'CHECK_IN_ANSWER',
  ANNOUNCEMENT = 'ANNOUNCEMENT',
  OPPORTUNITY = 'OPPORTUNITY',
}

export enum Roles {
  ENTITY_ADMIN = 'ENTITY_ADMIN',
  SALES_ADMIN = 'SALES_ADMIN',
  SYSTEM_ADMIN = 'SYSTEM_ADMIN',
  TEACHER = 'TEACHER',
}

export enum EducationalSettingTypes {
  HIGH_SCHOOL = 'HIGH_SCHOOL',
  MIDDLE_SCHOOL = 'MIDDLE_SCHOOL',
}

export enum ReportLevels {
  USER = 'USER',
  SCHOOL_CLASS = 'SCHOOL_CLASS',
  ENTITY = 'ENTITY',
}

export enum UploadReportStatuses {
  NOT_STARTED = 'NOT_STARTED',
  IN_PROGRESS = 'IN_PROGRESS',
  COMPLETED = 'COMPLETED',
  FAILED = 'FAILED',
}

export enum ArchivableStatusTypes {
  ACTIVE = 'ACTIVE',
  ALL = 'ALL',
  ARCHIVED = 'ARCHIVED',
}

export enum ContentStatusesTypes {
  DRAFT = 'DRAFT',
  PUBLISHED = 'PUBLISHED',
}

export enum ServiceTypes {
  ALL = 'ALL',
  CAREERS = 'CAREERS',
  LEARNING = 'LEARNING',
}

export enum AssignmentSubmissionStatus {
  DRAFT = 'DRAFT',
  SUBMITTED = 'SUBMITTED',
}

export enum OPPORTUNITY_APPLICATION_STATUS {
  ACCEPTED = 'ACCEPTED',
  FINISHED = 'FINISHED',
  PENDING = 'PENDING',
  REJECTED = 'REJECTED',
  STARTED = 'STARTED',
  EXPIRED = 'EXPIRED',
}

export enum VIRTUAL_INTERNSHIP_STATUS {
  NOT_STARTED = 'NOT_STARTED',
  IN_PROGRESS = 'IN_PROGRESS',
  COMPLETED = 'COMPLETED',
}

export enum OPPORTUNITY_TYPE {
  APPRENTICESHIP = 'APPRENTICESHIP',
  CLINICAL_EXPERIENCE = 'CLINICAL_EXPERIENCE',
  INTERNSHIP = 'INTERNSHIP',
  JOB_SHADOW = 'JOB_SHADOW',
  OTHER = 'OTHER',
  PRACTICUM = 'PRACTICUM',
  PRE_APPRENTICESHIP = 'PRE_APPRENTICESHIP',
  VIRTUAL_INTERNSHIP = 'VIRTUAL_INTERNSHIP',
  WORKPLACE_TOUR = 'WORKPLACE_TOUR',
  JOB = 'JOB',
  COLLEGE_VISIT = 'COLLEGE_VISIT',
  SPEAKER = 'SPEAKER',
}

export enum VISIBILITY_SCOPE {
  ALL = 'ALL',
  ENTITY = 'ENTITY',
}

export enum INSTITUTION_TYPES {
  PRV_NFP_LT2 = 'Private not-for-profit, less-than 2-year',
  PRV_NFP_2 = 'Private not-for-profit, 2-year',
  PRV_NFP_4 = 'Private not-for-profit, 4-year or above',
  PRV_FP_LT2 = 'Private for-profit, less-than 2-year',
  PRV_FP_2 = 'Private for-profit, 2-year',
  PRV_FP_4 = 'Private for-profit, 4-year or above',
  PUB_LT2 = 'Public, less-than 2-year',
  PUB_2 = 'Public, 2-year',
  PUB_4 = 'Public, 4-year or above',
}

export enum US_STATES {
  ALASKA = 'Alaska',
  ALABAMA = 'Alabama',
  ARKANSAS = 'Arkansas',
  AMERICAN_SAMOA = 'American Samoa',
  ARIZONA = 'Arizona',
  CALIFORNIA = 'California',
  COLORADO = 'Colorado',
  CONNECTICUT = 'Connecticut',
  DISTRICT_OF_COLUMBIA = 'District Of Columbia',
  DELAWARE = 'Delaware',
  FLORIDA = 'Florida',
  GEORGIA = 'Georgia',
  GUAM = 'Guam',
  HAWAII = 'Hawaii',
  IOWA = 'Iowa',
  IDAHO = 'Idaho',
  ILLINOIS = 'Illinois',
  INDIANA = 'Indiana',
  KANSAS = 'Kansas',
  KENTUCKY = 'Kentucky',
  LOUISIANA = 'Louisiana',
  MASSACHUSETTS = 'Massachusetts',
  MARYLAND = 'Maryland',
  MAINE = 'Maine',
  MICHIGAN = 'Michigan',
  MINNESOTA = 'Minnesota',
  MISSOURI = 'Missouri',
  MISSISSIPPI = 'Mississippi',
  MONTANA = 'Montana',
  NORTH_CAROLINA = 'North Carolina',
  NORTH_DAKOTA = 'North Dakota',
  NEBRASKA = 'Nebraska',
  NEW_HAMPSHIRE = 'New Hampshire',
  NEW_JERSEY = 'New Jersey',
  NEW_MEXICO = 'New Mexico',
  NEVADA = 'Nevada',
  NEW_YORK = 'New York',
  OHIO = 'Ohio',
  OKLAHOMA = 'Oklahoma',
  OREGON = 'Oregon',
  PENNSYLVANIA = 'Pennsylvania',
  PUERTO_RICO = 'Puerto Rico',
  RHODE_ISLAND = 'Rhode Island',
  SOUTH_CAROLINA = 'South Carolina',
  SOUTH_DAKOTA = 'South Dakota',
  TENNESSEE = 'Tennessee',
  TEXAS = 'Texas',
  UTAH = 'Utah',
  VIRGINIA = 'Virginia',
  VIRGIN_ISLANDS = 'Virgin Islands',
  VERMONT = 'Vermont',
  WASHINGTON = 'Washington',
  WISCONSIN = 'Wisconsin',
  WEST_VIRGINIA = 'West Virginia',
  WYOMING = 'Wyoming',
}

export enum APPLICATIONS_TYPE {
  COMMON_APP = 'COMMON_APP',
  DIRECT = 'DIRECT',
}

export enum RECOMMENDER_TYPE {
  TEACHER = 'TEACHER',
  COUNSELOR = 'COUNSELOR',
}

export enum INSTITUTION_APPLICATION_STATUS {
  NOT_STARTED = 'NOT_STARTED',
  IN_PROGRESS = 'IN_PROGRESS',
  SUBMITTED = 'SUBMITTED',
  COMPLETED = 'COMPLETED',
  DOWNLOADED = 'DOWNLOADED',
}
export enum COMMON_APP_FORM_STATUS {
  NOT_STARTED = 'NOT_STARTED',
  IN_PROGRESS = 'IN_PROGRESS',
  SUBMITTED = 'SUBMITTED',
  COMPLETED = 'COMPLETED',
  DOWNLOADED = 'DOWNLOADED',
}

export enum APPLICATION_FORM_TYPE {
  COUNSELOR_EARLY_DECISION_2 = 'Counselor early decision 2',
  COUNSELOR_EARLY_DECISION = 'Counselor early decision',
  COUNSELOR_FEE_WAIVER = 'Counselor fee waiver',
  COUNSELOR_FINAL_REPORT = 'Counselor final report',
  COUNSELOR_MIDYEAR_REPORT = 'Counselor midyear report',
  COUNSELOR_OPTIONAL_REPORT_2 = 'Counselor optional report 2',
  COUNSELOR_OPTIONAL_REPORT = 'Counselor optional report',
  COUNSELOR_PROFILE = 'Counselor profile',
  COUNSELOR_RECOMMENDATION = 'Counselor recommendation',
  COUNSELOR_SCHOOL_REPORT = 'Counselor school report',
  COUNSELOR_SECONDARY_REPORT = 'Counselor secondary report',
  TEACHER_PROFILE = 'Teacher profile',
  TEACHER_RECOMMENDATION = 'Teacher recommendation',
  TEACHER_EVALUATION = 'Teacher evaluation',
}

export enum SIZE_TYPES {
  VERY_SMALL = 'VERY_SMALL', // Size for very small institutions
  SMALL = 'SMALL', // Size for small institutions
  MEDIUM = 'MEDIUM', // Size for medium institutions
  LARGE = 'LARGE', // Size for large institutions
  VERY_LARGE = 'VERY_LARGE', // Size for very large institutions
}

export enum COST_RANGES {
  LESS_THAN_5000 = 'LESS_THAN_5000', // Institution cost value between 0 and 5000
  FROM_5001_TO_10000 = 'FROM_5001_TO_10000', //Institution cost value between 5001 and 10000
  FROM_10001_TO_15000 = 'FROM_10001_TO_15000', //Institution cost value between 10001 and 15000
  FROM_15001_TO_20000 = 'FROM_15001_TO_20000', //Institution cost value between 15001 and 20000
  MORE_THAN_20000 = 'MORE_THAN_20000', //Institution cost value higher than 20000
}

export enum LOCALE_STORAGE_KEY {
  APPLICATION_REFRESH_TIME = 'APPLICATION_REFRESH_TIME',
}

export enum CAREER_REVIEW_SURVEY_ANSWER_CONTEXT_TYPES {
  COURSE = 'COURSE',
  VIRTUAL_INTERNSHIP = 'VIRTUAL_INTERNSHIP',
  ASSESSMENT = 'ASSESSMENT',
}

export enum SYNC_STATUS {
  COMPLETED = 'COMPLETED',
  IN_PROGRESS = 'IN_PROGRESS',
  FAILED = 'FAILED',
}

export enum ASSIGNMENT_SUBMISSION_TYPES {
  COURSE = 'COURSE',
  VIRTUAL_INTERNSHIP = 'VIRTUAL_INTERNSHIP',
}

export enum ASSESSMENT_TYPES {
  HIGH_SCHOOL = 'HIGH_SCHOOL',
  MIDDLE_SCHOOL = 'MIDDLE_SCHOOL',
}

export enum ASSESSMENT_ATTEMPT_STATUS {
  IN_PROGRESS = 'IN_PROGRESS',
  FINISHED = 'FINISHED',
  REJECTED = 'REJECTED',
  PROCESSING_RESULTS = 'PROCESSING_RESULTS',
  FAILED = 'FAILED',
}

export enum ASSESSMENT_STATUSES {
  COMPLETED = 'COMPLETED',
  NOT_STARTED = 'NOT_STARTED',
  IN_PROGRESS = 'IN_PROGRESS',
}
