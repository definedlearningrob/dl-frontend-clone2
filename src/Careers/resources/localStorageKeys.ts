export const LOCAL_STORAGE_COURSE_ID = 'course-id';
