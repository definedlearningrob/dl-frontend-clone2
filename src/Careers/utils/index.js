const isObject = (object) => object !== null && typeof object === 'object';

export const isEqual = (object1, object2) => {
  if (object1 === object2) return true;

  if (
    object1 === null ||
    object2 === null ||
    typeof object1 !== 'object' ||
    typeof object2 !== 'object'
  )
    return false;

  const keys1 = Object.keys(object1);
  const keys2 = Object.keys(object2);

  if (keys1.length !== keys2.length) {
    return false;
  }

  for (let key of keys1) {
    const val1 = object1[key];
    const val2 = object2[key];
    const areObjects = isObject(val1) && isObject(val2);
    if (
      // eslint-disable-next-line
      areObjects && !isEqual(val1, val2) ||
      // eslint-disable-next-line
      !areObjects && val1 !== val2
    ) {
      return false;
    }
  }

  return true;
};

export const getByStringKey = (object, key) => {
  const keys = key.split('.');

  return keys.reduce((acc = {}, key) => acc[key], object);
};
