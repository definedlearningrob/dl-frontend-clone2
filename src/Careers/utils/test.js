/* eslint-disable react/prop-types */
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { createMemoryHistory } from 'history';
import { I18nextProvider } from 'react-i18next';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { Router, Route } from 'react-router-dom';

import i18n from '@dc/i18n';

export function renderWithRouter(
  ui,
  { route = '/', routePath = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {}
) {
  const Wrapper = ({ children }) => (
    <Router history={history}>
      <I18nextProvider i18n={i18n}>
        <Route path={routePath} render={() => children} />
      </I18nextProvider>
    </Router>
  );

  return {
    ...render(ui, { wrapper: Wrapper }),
    history,
  };
}

export function renderWithReduxProvider(
  ui,
  {
    initialState = {
      session: {
        loginError: {},
        user: null,
      },
    },
    store = configureMockStore([thunk])(initialState),
  } = {}
) {
  const Wrapper = ({ children }) => (
    <Provider store={store}>
      <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
    </Provider>
  );

  return {
    ...render(ui, { wrapper: Wrapper }),
  };
}

export function renderWithRouterAndReduxProvider(
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
    initialState = {
      session: {
        loginError: {},
        user: {},
      },
    },
    store = configureMockStore([thunk])(initialState),
  } = {},
  ...options
) {
  const Wrapper = ({ children }) => (
    <Provider store={store}>
      <Router history={history}>
        <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
      </Router>
    </Provider>
  );

  return {
    ...render(ui, { wrapper: Wrapper, ...options }),
    history,
  };
}

export function renderWithI18N(ui) {
  const Wrapper = ({ children }) => <I18nextProvider i18n={i18n}>{children}</I18nextProvider>;

  return {
    ...render(ui, { wrapper: Wrapper }),
  };
}
