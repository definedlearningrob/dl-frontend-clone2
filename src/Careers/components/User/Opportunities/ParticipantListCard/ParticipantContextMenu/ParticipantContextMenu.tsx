import * as DropdownMenu from '@radix-ui/react-dropdown-menu';
import { useMemo, useCallback, FC, SVGProps } from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory, useParams } from 'react-router-dom';

import { OPPORTUNITY_APPLICATION_STATUS } from '@dc/resources/enums';
import { CONVERSATION_CONTEXT_TYPES } from '@dc/resources/constants';

import { ReactComponent as FlagIcon } from '@shared/assets/icons/flag_outlined.svg';
import { ReactComponent as DotsIcon } from '@shared/svg/three_dots.svg';
import { ReactComponent as SendIcon } from '@shared/svg/send_outlined.svg';
import { ReactComponent as UserIcon } from '@shared/svg/user_outlined.svg';
import { useMessaging } from '@shared/hooks/useMessaging';
import { RECEIVER_TYPES } from '@shared/resources/constants';
import { IconButton } from '@shared/components/IconButton/IconButton';
import Button from '@shared/components/Button/Button';
import { cx } from '@shared/utils/cx';

type ContextMenuOption = {
  hide?: boolean;
  Icon: FC<SVGProps<SVGSVGElement>>;
  label: string;
  onClick: () => void;
};

type Props = {
  setUpdateStatusModalOpen: (options: {
    applicationId: string;
    applicationStatus: OPPORTUNITY_APPLICATION_STATUS;
    fullName: string;
  }) => void;
  student: {
    applicationId: string;
    applicationStatus: OPPORTUNITY_APPLICATION_STATUS;
    fullName: string;
    uuid: string;
  };
  opportunityName: string;
  isOpen: boolean;
  toggleOpen: () => void;
};

export const ParticipantContextMenu = ({
  setUpdateStatusModalOpen,
  student,
  opportunityName,
  isOpen,
  toggleOpen,
}: Props) => {
  const { uuid: studentUuid, fullName, applicationId, applicationStatus } = student;
  const { id } = useParams<{ id: string }>();
  const { messagingState, setMessagingState } = useMessaging();
  const { t } = useTranslation();
  const history = useHistory();
  const hideStatusUpdateOption = ![
    OPPORTUNITY_APPLICATION_STATUS.ACCEPTED,
    OPPORTUNITY_APPLICATION_STATUS.FINISHED,
    OPPORTUNITY_APPLICATION_STATUS.STARTED,
  ].includes(applicationStatus);

  const handleSendMessage = useCallback(() => {
    setMessagingState({
      ...messagingState,
      show: true,
      receiverType: RECEIVER_TYPES.STUDENT,
      actionContext: {
        type: CONVERSATION_CONTEXT_TYPES.OPPORTUNITY,
        id,
        title: opportunityName,
      },
      receiver: { ...student, name: fullName },
    });
  }, []);

  const handleShowPortfolio = () => {
    history.push(`/students/${studentUuid}/portfolio`);
  };

  const handleSetUpdateStatusModalOpen = () =>
    setUpdateStatusModalOpen({ applicationId, applicationStatus, fullName });

  const optionStatusLabel = useMemo(
    () =>
      ({
        [OPPORTUNITY_APPLICATION_STATUS.ACCEPTED]: t('user.opportunities.contextMenu.inProgress'),
        [OPPORTUNITY_APPLICATION_STATUS.STARTED]: t('user.opportunities.contextMenu.completed'),
        [OPPORTUNITY_APPLICATION_STATUS.FINISHED]: t('user.opportunities.contextMenu.inProgress'),
        [OPPORTUNITY_APPLICATION_STATUS.PENDING]: null,
        [OPPORTUNITY_APPLICATION_STATUS.REJECTED]: null,
        [OPPORTUNITY_APPLICATION_STATUS.EXPIRED]: null,
      }[applicationStatus]),
    [applicationStatus]
  );

  const contextMenuOptions: ContextMenuOption[] = [
    {
      Icon: UserIcon,
      label: t('user.opportunities.contextMenu.showPortfolio'),
      onClick: handleShowPortfolio,
    },
    {
      Icon: SendIcon,
      label: t('user.opportunities.contextMenu.sendMessage'),
      onClick: handleSendMessage,
    },
    {
      hide: hideStatusUpdateOption,
      Icon: FlagIcon,
      label: t('user.opportunities.contextMenu.setStatus', {
        status: optionStatusLabel,
      }),
      onClick: handleSetUpdateStatusModalOpen,
    },
  ].filter((option) => !option.hide);

  return (
    <DropdownMenu.Root modal={false} open={isOpen} onOpenChange={toggleOpen}>
      <DropdownMenu.Trigger asChild={true}>
        <IconButton
          Icon={DotsIcon}
          className={cx('invisible group-hover:visible hover:bg-white', {
            'visible outline outline-1 outline-primary-500': isOpen,
          })}
          variant='white'
        />
      </DropdownMenu.Trigger>
      <DropdownMenu.Portal>
        <DropdownMenu.Content align='end' sideOffset={8}>
          <ul className='rounded-sm bg-white border border-neutral-300'>
            {contextMenuOptions.map((option) => (
              <li key={option.label} className='flex'>
                <Button
                  Icon={option.Icon}
                  className='w-full !justify-start font-regular'
                  size='sm'
                  onClick={option.onClick}>
                  {option.label}
                </Button>
              </li>
            ))}
          </ul>
        </DropdownMenu.Content>
      </DropdownMenu.Portal>
    </DropdownMenu.Root>
  );
};
