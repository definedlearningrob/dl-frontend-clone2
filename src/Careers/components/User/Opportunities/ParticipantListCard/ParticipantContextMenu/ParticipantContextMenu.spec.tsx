import userEvent from '@testing-library/user-event';
import { screen } from '@testing-library/dom';
import { createMemoryHistory } from 'history';

import { renderWithRouterAndReduxProvider } from '@dc/utils/test';
import { ParticipantContextMenu } from '@dc/components/User/Opportunities/ParticipantListCard/ParticipantContextMenu/ParticipantContextMenu';
import { OPPORTUNITY_APPLICATION_STATUS } from '@dc/resources/enums';

const statusChangeSpy = jest.fn();

const getStudentMock = (applicationStatus: OPPORTUNITY_APPLICATION_STATUS) => ({
  applicationId: '1',
  applicationStatus,
  fullName: 'Student Name',
  uuid: '123',
});

describe('ParticipantContextMenu', () => {
  it('should render correctly', () => {
    const { container } = renderWithRouterAndReduxProvider(
      <ParticipantContextMenu
        isOpen={true}
        opportunityName='Opportunity name'
        setUpdateStatusModalOpen={jest.fn()}
        student={getStudentMock(OPPORTUNITY_APPLICATION_STATUS.ACCEPTED)}
        toggleOpen={() => {}}
      />
    );

    expect(container).toMatchSnapshot();
  });

  describe('should have proper options for', () => {
    it('Not Accepted application', async () => {
      renderWithRouterAndReduxProvider(
        <div id='portal'>
          <ParticipantContextMenu
            isOpen={true}
            opportunityName='Opportunity name'
            setUpdateStatusModalOpen={jest.fn()}
            student={getStudentMock(OPPORTUNITY_APPLICATION_STATUS.REJECTED)}
            toggleOpen={() => {}}
          />
        </div>
      );

      const items = await screen
        .findAllByRole('listitem')
        .then((elements) => elements.map((item) => item.textContent));

      expect(items).toEqual(['Show Portfolio', 'Send Message']);
    });

    it('Accepted application', async () => {
      renderWithRouterAndReduxProvider(
        <div id='portal'>
          <ParticipantContextMenu
            isOpen={true}
            opportunityName='Opportunity name'
            setUpdateStatusModalOpen={jest.fn()}
            student={getStudentMock(OPPORTUNITY_APPLICATION_STATUS.ACCEPTED)}
            toggleOpen={() => {}}
          />
        </div>
      );

      const items = await screen
        .findAllByRole('listitem')
        .then((elements) => elements.map((item) => item.textContent));

      expect(items).toEqual(['Show Portfolio', 'Send Message', 'Set as In Progress']);
    });

    it('In-Progress application', async () => {
      renderWithRouterAndReduxProvider(
        <div id='portal'>
          <ParticipantContextMenu
            isOpen={true}
            opportunityName='Opportunity name'
            setUpdateStatusModalOpen={jest.fn()}
            student={getStudentMock(OPPORTUNITY_APPLICATION_STATUS.STARTED)}
            toggleOpen={() => {}}
          />
        </div>
      );

      const items = await screen
        .findAllByRole('listitem')
        .then((elements) => elements.map((item) => item.textContent));

      expect(items).toEqual(['Show Portfolio', 'Send Message', 'Set as Completed']);
    });

    it('Completed application', async () => {
      renderWithRouterAndReduxProvider(
        <div id='portal'>
          <ParticipantContextMenu
            isOpen={true}
            opportunityName='Opportunity name'
            setUpdateStatusModalOpen={jest.fn()}
            student={getStudentMock(OPPORTUNITY_APPLICATION_STATUS.FINISHED)}
            toggleOpen={() => {}}
          />
        </div>
      );

      const items = await screen
        .findAllByRole('listitem')
        .then((elements) => elements.map((item) => item.textContent));

      expect(items).toEqual(['Show Portfolio', 'Send Message', 'Set as In Progress']);
    });
  });

  describe('should allow to change status to', () => {
    it('in-progress', async () => {
      renderWithRouterAndReduxProvider(
        <div id='portal'>
          <ParticipantContextMenu
            isOpen={true}
            opportunityName='Opportunity name'
            setUpdateStatusModalOpen={statusChangeSpy}
            student={{
              applicationId: '1',
              applicationStatus: OPPORTUNITY_APPLICATION_STATUS.ACCEPTED,
              fullName: 'Student Name',
              uuid: '123',
            }}
            toggleOpen={() => {}}
          />
        </div>
      );

      const progressChangeItem = await screen.findByText('Set as In Progress');
      userEvent.click(progressChangeItem);

      expect(statusChangeSpy).toHaveBeenCalledWith({
        applicationId: '1',
        applicationStatus: 'ACCEPTED',
        fullName: 'Student Name',
      });
    });

    it('completed', async () => {
      renderWithRouterAndReduxProvider(
        <div id='portal'>
          <ParticipantContextMenu
            isOpen={true}
            opportunityName='Opportunity name'
            setUpdateStatusModalOpen={statusChangeSpy}
            student={{
              applicationId: '1',
              applicationStatus: OPPORTUNITY_APPLICATION_STATUS.STARTED,
              fullName: 'Student Name',
              uuid: '123',
            }}
            toggleOpen={() => {}}
          />
        </div>
      );

      const progressChangeItem = await screen.findByText('Set as Completed');
      userEvent.click(progressChangeItem);

      expect(statusChangeSpy).toHaveBeenCalledWith({
        applicationId: '1',
        applicationStatus: 'STARTED',
        fullName: 'Student Name',
      });
    });
  });

  it('should allow to navigate to student portfolio', async () => {
    const history = createMemoryHistory({ initialEntries: ['/opportunities/1'] });

    renderWithRouterAndReduxProvider(
      <div id='portal'>
        <ParticipantContextMenu
          isOpen={true}
          opportunityName='Opportunity name'
          setUpdateStatusModalOpen={() => {}}
          student={{
            applicationId: '1',
            applicationStatus: OPPORTUNITY_APPLICATION_STATUS.ACCEPTED,
            fullName: 'Student Name',
            uuid: '123',
          }}
          toggleOpen={() => {}}
        />
      </div>,
      { history }
    );

    const showPortfolioItem = await screen.findByText('Show Portfolio');
    userEvent.click(showPortfolioItem);

    expect(history.location.pathname).toEqual('/students/123/portfolio');
  });
});
