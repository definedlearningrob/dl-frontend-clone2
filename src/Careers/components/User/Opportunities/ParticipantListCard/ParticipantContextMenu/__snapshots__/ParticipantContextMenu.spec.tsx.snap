// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParticipantContextMenu should render correctly 1`] = `
<div>
  <button
    aria-controls="radix-1"
    aria-expanded="true"
    aria-haspopup="menu"
    class="flex items-center justify-center transition-colors rounded-sm border border-transparent cursor-pointer focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] text-primary-500 bg-white hover:bg-primary-200 h-base w-base  group-hover:visible hover:bg-white visible outline outline-1 outline-primary-500"
    data-state="open"
    data-testid="icon-button"
    id="radix-0"
    type="button"
  >
    <div
      class="flex"
    >
      <div
        class="flex align-center justify-center grow-0 shrink-0"
        data-testid="icon"
      >
        <div
          class="h-sm w-sm"
        />
      </div>
    </div>
  </button>
</div>
`;
